### PR TITLE
feat: added notify button experiment

### PR DESCRIPTION
**Pull request by ilasw**
Part 2/2 of https://github.com/dailydotdev/apps/pull/3360

## Events

Did you introduce any new tracking events?

| Type   | event_name  | value |
|--------|-------------|-------|
| Source follow| follow  | extra: { ... } |
| Source unfollow| unfollow  | extra: { ... } |
| Source get notifications| subscribe  | extra: { ... } |
| Source remove notifications| unsubscribe  | extra: { ... } |
| Source unblock | ~~follow~~ unblock  | extra: { ... } |

## Experiment

Introduced `sourceNotifyButton` experiment

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://source-actions-notify-button.preview.app.daily.dev

test in source page: https://source-actions-notify-button.preview.app.daily.dev/sources/codrops
test in article page: https://source-actions-notify-button.preview.app.daily.dev/posts/creating-a-glowing-text-marquee-animation-2vkiskfrm

### CODE REVIEW

**Comment by nensidosari** [comment]
Just referencing this conversation on events here: https://github.com/dailydotdev/apps/pull/3383#discussion_r1687818057 since this is where any possible change will be done

File: packages/shared/src/hooks/useTagAndSource.ts
Comment (idoshamun) [inline packages/shared/src/hooks/useTagAndSource.ts:213]: let's fix that to unblock
```suggestion
        event_name: 'unblock',
```

File: packages/shared/src/hooks/useTagAndSource.ts
Comment (idoshamun) [inline packages/shared/src/hooks/useTagAndSource.ts:274]:
```suggestion
        event_name: 'follow',
```

File: packages/shared/src/hooks/useTagAndSource.ts
Comment (idoshamun) [inline packages/shared/src/hooks/useTagAndSource.ts:305]:
```suggestion
        event_name: 'unfollow',
```

File: packages/shared/src/components/PostOptionsMenu.tsx
Comment (ilasw) [inline packages/shared/src/components/PostOptionsMenu.tsx:126]: useConditionalFeature check is open

Comment (capJavert) [inline packages/shared/src/components/PostOptionsMenu.tsx:126]: `useConditionalFeature` only when options is open

**Review by nensidosari — APPROVED** [review]
PR looks great, changes are super clear 🚀 . 
Approving since my comments are non blocking just on my opinion make code clearer and remove the need for comments. 

Last reminder, let's not forget to send a message to experiments channel about experiment enrolment and waiting for Ido's approval there before we merge the PR.  🙏

File: packages/shared/src/hooks/source/useSourceActions.ts
Comment (rebelchris) [inline packages/shared/src/hooks/source/useSourceActions.ts:15]: Im not sure of the name is clear enough, it sounds like user has notifications open.
I think you want to say when someone has notify on right?
Not sure whats better but feel this naming might give confusion down the line (cc @nensidosari what do you feel?)

**Review by rebelchris — APPROVED** [review]
Just one remark for future proofing 👍

File: packages/shared/src/hooks/source/useSourceActions.ts
Comment (nensidosari) [reply packages/shared/src/hooks/source/useSourceActions.ts:15]: Yeah it's a bit confusing, maybe `haveNotificationsOn` is really better.

File: packages/shared/src/components/sources/SourceActions/SourceActionsFollow.tsx
Comment (ilasw) [reply packages/shared/src/components/sources/SourceActions/SourceActionsFollow.tsx:18]: 👍

File: packages/shared/src/components/sources/SourceActions/SourceActionsWithNotifyExperiment.tsx
Comment (ilasw) [reply packages/shared/src/components/sources/SourceActions/SourceActionsWithNotifyExperiment.tsx:16]: ✅

File: packages/shared/src/components/PostOptionsMenu.tsx
Comment (nensidosari) [inline packages/shared/src/components/PostOptionsMenu.tsx:368]: since this is destructured above
```suggestion
             haveNotificationsOn
```

File: packages/shared/src/components/PostOptionsMenu.tsx
Comment (nensidosari) [inline packages/shared/src/components/PostOptionsMenu.tsx:375]: since this is destructured above

```suggestion
         haveNotificationsOn
```

**Review by nensidosari — APPROVED** [review]
Looks great

**Comment by nensidosari** [comment]
All issues mentioned in [comment above](https://github.com/dailydotdev/apps/pull/3385#issuecomment-2248041432) are now resolved! Thank u @ilasw

File: packages/shared/src/lib/log.ts
Comment (capJavert) [inline packages/shared/src/lib/log.ts:158]: These 4 are not used anywhere in this PR from what I see in the diff?

Comment (capJavert) [reply packages/shared/src/lib/log.ts:158]: Did you agree somewhere else on this? https://github.com/dailydotdev/apps/pull/3385#discussion_r1689150558

**Review by capJavert — APPROVED** [review]
Sanity checking for events, don't see anything else code wise 👍

File: packages/shared/src/lib/log.ts
Comment (ilasw) [reply packages/shared/src/lib/log.ts:158]: They where used since last pull request, now I've changed the values to reflect the new introduced events 👀

**Pull request merged**
